packages/confirm-dialog/src/vaadin-confirm-dialog-dialog.js

```diff
@@ -0,0 +1,112 @@
+ */
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { Dialog, DialogOverlay } from '@vaadin/dialog/src/vaadin-dialog.js';
+import { css, registerStyles } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+
+registerStyles(
+  'vaadin-confirm-dialog-overlay',
+  css`
+    :host {
+      --_vaadin-confirm-dialog-content-width: auto;
+      --_vaadin-confirm-dialog-content-height: auto;
+      --_vaadin-confirm-dialog-footer-height: auto;
+    }
+
+    [part='content'] {
+      width: var(--_vaadin-confirm-dialog-content-width);
+      height: auto;
+      box-sizing: content-box;
+    }
+  `,
+  { moduleId: 'vaadin-confirm-dialog-overlay-styles' }
+);
+
+let memoizedTemplate;
+
+const dialogTemplate = html`
+  <div part="container">
+    <div part="header">
+      <slot name="header"></slot>
+    </div>
+
+    <div part="message">
+      <slot></slot>
+    </div>
+  </div>
+
+  <div part="footer">
+    <div part="cancel-button">
+      <slot name="cancel-button"></slot>
+    </div>
+    <div part="reject-button">
+      <slot name="reject-button"></slot>
+    </div>
+    <div part="confirm-button">
+      <slot name="confirm-button"></slot>
+    </div>
+  </div>
+`;
+
+class ConfirmDialogOverlay extends DialogOverlay {
+  static get is() {
+    return 'vaadin-confirm-dialog-overlay';
+  }
+
+  static get template() {
+    if (!memoizedTemplate) {
+      memoizedTemplate = super.template.cloneNode(true);
+      const contentPart = memoizedTemplate.content.querySelector('[part="content"]');
+      const defaultSlot = contentPart.querySelector('slot:not([name])');
+      contentPart.removeChild(defaultSlot);
+      contentPart.appendChild(dialogTemplate.content.cloneNode(true));
+    }
+    return memoizedTemplate;
+  }
+
+  /**
+   * Override method inherited from `OverlayElement` to notify when overlay is closed.
+   * The `vaadin-overlay-close` event is not suitable, as it fires before closing.
+   * @protected
+   * @override
+   */
+  _finishClosing() {
+    super._finishClosing();
+
+    this.dispatchEvent(new CustomEvent('vaadin-confirm-dialog-close'));
+  }
+}
+
+customElements.define('vaadin-confirm-dialog-overlay', ConfirmDialogOverlay);
+
+class ConfirmDialogDialog extends Dialog {
+  /**
+   * Override template to provide custom overlay tag name.
+   */
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: none;
+        }
+      </style>
+
+      <vaadin-confirm-dialog-overlay
+        id="overlay"
+        on-opened-changed="_onOverlayOpened"
+        on-mousedown="_bringOverlayToFront"
+        on-touchstart="_bringOverlayToFront"
+        theme$="[[theme]]"
+        modeless="[[modeless]]"
+        with-backdrop="[[!modeless]]"
+        resizable$="[[resizable]]"
+        focus-trap
+      ></vaadin-confirm-dialog-overlay>
+    `;
+  }
+}
+
+customElements.define('vaadin-confirm-dialog-dialog', ConfirmDialogDialog);
```

Comment:
I kinda got confused with this name.. looks like a copy paste error, or? 😅

Reply:
We already have `vasdin-time-picker-time-picker` 🙈 

How about if I change it to be `vaadin-confirn-dialog-internal-dialog`, would it make it make it less confusing?

Reply:
Oh! Good question - not sure if renaming or adding a comment above the define would be better for other people :)

Reply:
Updated the definition for internal custom elements, added corresponding JSDoc annotations.